hatchways/team-evergreen
# Mongo db Configuration

This commit creates a configuration file that uses environment variables to define the location of your mongo DB as well as the names of the dev and test databases.  I went with evergreen_dev and evergreen_test but you can override the defaults by setting your own .env variables.

**Pre-Requisites:**
[Install MongoDB](https://docs.mongodb.com/manual/installation/) on your machine by following the instructions for your os type.  My installation was for version 4.2.

**To test on your machine:**
1. Go to app\server\ directory
2.  Download the new code - `git pull origin MongoDB`
3. Run ` npm install` to updated required packages (mongoose and dotenv)
4. From terminal start mongoDB - (script for Mac can be run by running `npm run start-mongo-db-on-mac` - if you've installed it in the standard directory, otherwise follow instructions for your os\build).  A successful launch of MongoDB will leave you with a hung terminal window with no status message - if you see errors fix them ... :-). 
5. In a new terminal windows go to app\server directory.
6. Run `node run test`.  This will run the unit tests for the model file and the db-connect file.  All 5 tests should pass.

Some additional tools, I downloaded compass which is a gui for using MongoDB.  You can find download instructions [here ](https://www.mongodb.com/products/compass).  Download the community version.

**Additional Notes**
1.  The default install of MongoDB does not require any access permissions or authorization - I find this disturbing but didn't want to mess with it - but I figure at some point we will want to have some sort of authorization to the database.
2. I look forward to the day where writing 10 lines of code does not require 6 hours of research :-).
